### PR TITLE
Node pathway/Linting: Add section on template repos

### LINKED ISSUE #26515: Webpack: Add mention of template repos
### Describe your suggestion

The webpack lesson has very recently been updated but one extra thing it could do with is a quick mention about template repos for future setup convenience. Occasionally people do question whether they need to do all of the config themselves from scratch every new project or "hack it" by copying and pasting locally, whereas a template repo with a basic configuration and file structure would be far more convenient.

I don't imagine this will be much work for someone to do.

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/node-path-javascript-webpack

### Checks

- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### (Optional) Discord Name

Mao

### (Optional) Additional Comments

_No response_

### PR DESCRIPTION
**Pull request by MaoShizhong**
## Because
Template repos are simple and very handy, and are commonly suggested when learners ask about easier ways to manage things like reusing Webpack config for new projects.
A brief mention of what they are and how to set/use one has been added to the end of the Linting lesson, as part of the lesson's "productivity" theme, whilst also giving the learner an opportunity to get a better feel for Webpack across Restaurant Page/To-do list first.

## This PR
- Adds content on template repos
- Rephrase intro text to account for new content
- Add knowledge check and doc link for template repos

## Issue
Closes #26515

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: javascript/javascript_in_the_real_world/linting.md [hunk 1]
```diff
@@ -48,10 +49,19 @@ We **highly** recommend that you install ESlint and Prettier and use them for al
 However, using ESLint and Prettier together causes conflicts. To fix that follow the instructions to install [eslint-config-prettier](https://github.com/prettier/eslint-config-prettier#installation). It turns off all ESLint rules that are unnecessary or might conflict with Prettier. Doing just this is enough to resolve the conflict and get them both working smoothly with one another.
 Another way to address the conflict is to use `eslint-plugin-prettier`. It lets you run Prettier as if it were a rule in ESLint. However, doing this is **not recommended**. You can learn more about it [here](https://prettier.io/docs/en/integrating-with-linters.html#notes).
 
+### Template repositories
+
+With the last few projects, you have felt that setting up Webpack involved a fair few files and configuration, and that you may have had to look at what you configured before to copy and paste the configuration you want to reuse. You may also have noticed that whenever you create a new repository on Github, there is an option near the top for a `Repository template`.
```
Comment (wise-king-sullyman) [inline javascript/javascript_in_the_real_world/linting.md:54]:
```suggestion
With the last few projects, you might have felt that setting up Webpack involved a fair few files and configuration, and that you may have had to look at what you configured before to copy and paste the configuration you want to reuse. You may also have noticed that whenever you create a new repository on Github, there is an option near the top for a `Repository template`.
```

**Review by wise-king-sullyman — CHANGES_REQUESTED** [review]
I dig it 🚀 

One small nit:

**Comment by MaoShizhong** [comment]
Good catch, it must have gotten chopped off in a reword and I missed it...thank you!
I also just noticed the branch spelling...

**Comment by wise-king-sullyman** [comment]
😂 I hadn't noticed the branch name until you mentioned it!

**Review by wise-king-sullyman — APPROVED** [review]
🔥 🔥 🔥

**Pull request merged**